TDL-09-shell-EC2/deploy_instance.sh

```diff
@@ -0,0 +1,229 @@
+#!/bin/bash
+#
+# Deploy AWS instance.
+# Author: Yvonne Yao
+
+export AWS_DEFAULT_REGION='ap-southeast-2'
+DEFAULT_TIME_ZONE='Australia/Melbourne'
+declare -a LIST_OSTYPE
+LIST_OSTYPE=('amzn' 'amzn2' 'centos6' 'centos7' 'ubuntu')
+declare -a LIST_INSTANCE_TYPE
+LIST_INSTANCE_TYPE=('t2.micro' 't2.small' 't3.micro')
+instance_type='t2.micro' #default value
+
+#######################################
+# Print a message in a given color.
+# Arguments:
+#   Color. eg: green, red
+#######################################
+function print_color(){
+  NC='\033[0m' # No Color
+  case $1 in
+    "green") COLOR='\033[0;32m' ;;
+    "red") COLOR='\033[0;31m' ;;
+    "*") COLOR='\033[0m' ;;
+  esac
+
+  echo -e "${COLOR} $2 ${NC}"
+}
+
+#######################################
+# Check item in a list.
+# Arguments:
+#   list item
+#######################################
+function list_include_item {
+  list="$1"
+  item="$2"
+  if [[ $list =~ (^|[[:space:]])"$item"($|[[:space:]]) ]] ; then
+    result=0  # yes, list include item
+  else
+    result=1
+  fi
+  echo $result
+  return $result
+}
+
+#######################################
+# Show how to use the deploy_instance.sh.
+# Arguments:
+#   usage_type. eg: all, ostype, agedtime, instance_type.
+#######################################
+function usage(){
+  case "$1" in
+    all)
+      print_color "red" "To deploy your instance, please run deploy_instance.sh as follows:"
+      print_color "red" "./deploy_instance.sh ostype agedtime [instance_type]"
+      print_color "red" "The value range of the arguments: "
+      print_color "red" "1. valid ostype: amzn, amzn2, centos6, centos7, ubuntu"
+      print_color "red" "2. agedtime: 1-7 "
+      print_color "red" "3. instance_type: default t2_micro, valid instance_type: t2.micro, t2.small, t3.micro"
+      ;;
```

Comment:
color red normally is used for error msg:)

Reply:
ok. Thanks.